Solutions/Ais.Net.Receiver/StorageClient.cs

```diff
@@ -15,33 +16,58 @@ namespace Endjin.Ais.Receiver
     using System.Text;
     using System.Threading.Tasks;
 
-    public class StorageClient
+    public class StorageClient : IStorageClient
     {
-        private AppendBlobClient container;
-        private IConfiguration configuration;
-        private Stream stream;
+        private readonly IConfiguration configuration;
+        private AppendBlobClient appendBlobClient;
+        private BlobContainerClient blobContainerClient;
 
         public StorageClient(IConfiguration configuration)
         {
             this.configuration = configuration;
         }
 
-        public async Task AppendMessages(IEnumerable<string> messages)
+        public async Task PersistAsync(IEnumerable<string> messages)
         {
-            await this.InitialiseConnectionAsync().ConfigureAwait(false);
+            await this.InitialiseContainerAsync().ConfigureAwait(false);
+            await using MemoryStream stream = new MemoryStream(Encoding.UTF8.GetBytes(messages.Aggregate(new StringBuilder(), (sb, a) => sb.AppendLine(string.Join(",", a)), sb => sb.ToString())));
+            await this.appendBlobClient.AppendBlockAsync(stream).ConfigureAwait(false);
+        }
 
-            using (var stream = new MemoryStream(Encoding.UTF8.GetBytes(messages.Aggregate(new StringBuilder(), (sb, a) => sb.AppendLine(String.Join(",", a)), sb => sb.ToString())))) 
+        private async Task InitialiseContainerAsync()
```

Comment:
I think this is misnamed. You call it every time `PersistAsync` is called, so this doesn't really look like an initialization method.